src/borg/repository.py

```diff
@@ -750,10 +750,53 @@ def __contains__(self, id):
         return id in self.index
 
     def list(self, limit=None, marker=None):
+        """
+        list <limit> IDs starting from after id <marker> - in index (pseudo-random) order.
+        """
         if not self.index:
             self.index = self.open_index(self.get_transaction_id())
         return [id_ for id_, _ in islice(self.index.iteritems(marker=marker), limit)]
 
+    def scan(self, limit=None, marker=None):
```

Comment:
about list vs. scan:

somehow this is the same thing, so just do both via list() and add a linear=False param to list? in borg 1.0.x we could then do it in the "try newcall / except oldcall" way as we did at other places.